_plugins/auto_logs_markup.rb

```diff
@@ -38,6 +48,7 @@
     time    = line[0..TIME_SIZE]
     name    = IRC_NICK.match(line).to_s
     nick    = name.gsub(LT_GT, '').strip
+    color   = name_colors[nick] || (name_colors[nick] = colors[color_index += 1])
```

Comment:
Note that the colors array is currently 63 elements in size; if we have more than that many participants in a post, the color will become `nil`. We could check that `color_index < colors.size` and if true reset the index to 0, or ensure the color array is large enough.